cmake-modules/AddTestCMocka.cmake

```diff
@@ -77,7 +77,21 @@ function(ADD_CMOCKA_TEST _TARGET_NAME)
     endif()
 
     if(DEFINED ENV{VCPKG_ROOT} OR DEFINED ENV{VCPKG_INSTALLATION_ROOT})
-        set(CMOCKA_LIB ${CMOCKA_LIBRARIES})
+        if(DEFINED ENV{VCPKG_ROOT})
+            set(VCPKG_PATH $ENV{VCPKG_ROOT})
+        endif()
+        if(DEFINED ENV{VCPKG_INSTALLATION_ROOT})
+            set(VCPKG_PATH "$ENV{VCPKG_INSTALLATION_ROOT}/")
+        endif()
+
+        # Patch for new VCPKG cmocka returning relative path
+        if(MSVC)
+            set(CMOCKA_STATIC_LIB "/lib/cmocka-static.lib")
+        else()
+            set(CMOCKA_STATIC_LIB "/lib/libcmocka-static.a")
+        endif()
+        set(CMOCKA_PREFIX "${VCPKG_PATH}installed/${VCPKG_TARGET_TRIPLET}")
+        set(CMOCKA_LIB "debug;${CMOCKA_PREFIX}/debug${CMOCKA_STATIC_LIB};optimized;${CMOCKA_PREFIX}${CMOCKA_STATIC_LIB}")
```

Comment:
Question: How did you know this is what we need to set `CMOCKA_LIB` to? Will we always have `debug`?

I am trying to understand what this means.

Reply:
no, 
this is how a package tells VCPKG where to find libraries for debug and for optimized.
The string looks like `degug;path;optimized;path`

Then, depending on what cmake is building, it will pick debug or optimized.

Then one on optimized does not have the `debug/`